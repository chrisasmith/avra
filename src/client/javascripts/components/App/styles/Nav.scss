@import "../../../../stylesheets/modules/theme.scss";

.navBar {
  height: 40px;
  color: $darker-gray-color;
  background-color: $nielsen-blue-color;

  a {
    color: $darker-gray-color;

    &:visited {
      color: $darker-gray-color;
    }

    &:hover {
      color: $white-color;
    }

    &.selected {
      color: $white-color;
    }
  }

  input {
    height: 30px;
  }
}
