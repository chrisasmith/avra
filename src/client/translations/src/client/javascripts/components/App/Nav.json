[
  {
    "id": "nav_bar.search",
    "defaultMessage": "Search"
  },
  {
    "id": "nav_bar.title",
    "defaultMessage": "AVRA"
  },
  {
    "id": "nav_bar.nav_your_fans",
    "defaultMessage": "Your Fans"
  },
  {
    "id": "nav_bar.nav_your_partners",
    "defaultMessage": "Your Partners"
  },
  {
    "id": "nav_bar.nav_find_partners",
    "defaultMessage": "Find Partners"
  }
]