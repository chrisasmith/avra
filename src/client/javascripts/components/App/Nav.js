import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import { Link } from 'react-router'
import { injectIntl, defineMessages, FormattedMessage } from 'react-intl'
import { routes } from '../../routes'
import styles from './styles/Nav.scss'

const propTypes = {
  location: PropTypes.object.isRequired,
}

const messages = defineMessages({
  searchPlaceholder: {
    id: 'nav_bar.search',
    defaultMessage: 'Search',
  },
})

const Nav = injectIntl(({ intl, location }) => (
  <nav className={classnames(['navbar', styles.navBar])}>
    <a className="navbar-brand" href="#">
      <FormattedMessage
        id="nav_bar.title"
        defaultMessage="AVRA"
      />
    </a>
    <Link className={classnames('nav-link', { [styles.selected]: location.pathname.startsWith(routes.YOURFANS.ROOT) })} to={routes.YOURFANS.ROOT}>
      <FormattedMessage
        id="nav_bar.nav_your_fans"
        defaultMessage="Your Fans"
      />
    </Link>
    <Link className={classnames('nav-link', { [styles.selected]: location.pathname.startsWith(routes.PARTNERS.ROOT) })} to={routes.PARTNERS.ROOT}>
      <FormattedMessage
        id="nav_bar.nav_your_partners"
        defaultMessage="Your Partners"
      />
    </Link>
    <Link className={classnames('nav-link', { [styles.selected]: location.pathname.startsWith(routes.FINDPARTNERS.ROOT) })} to={routes.FINDPARTNERS.ROOT}>
      <FormattedMessage
        id="nav_bar.nav_find_partners"
        defaultMessage="Find Partners"
      />
    </Link>
    <form className="form-inline">
      <input className="form-control form-control-lg" type="search" placeholder={intl.formatMessage(messages.searchPlaceholder)} aria-label="Search" />
    </form>
    <a className="nav-link" href="#">Jane Doe</a>
  </nav>
))

Nav.propTypes = propTypes
export default Nav
