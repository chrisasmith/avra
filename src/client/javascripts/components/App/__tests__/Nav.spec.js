import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Nav from '../Nav'

describe('App components nav', () => {
  it('should render', () => {
    const wrapper = shallow(<Nav location={{ pathname: '/' }} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
