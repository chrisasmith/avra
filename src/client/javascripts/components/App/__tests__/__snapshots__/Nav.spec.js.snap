// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components nav should render 1`] = `
<nav
  className="navbar navBar"
>
  <a
    className="navbar-brand"
    href="#"
  >
    AVRA
  </a>
  <Link
    className="nav-link"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/your_fans"
  >
    Your Fans
  </Link>
  <Link
    className="nav-link"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/your_partners"
  >
    Your Partners
  </Link>
  <Link
    className="nav-link"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/find_partners"
  >
    Find Partners
  </Link>
  <form
    className="form-inline"
  >
    <input
      aria-label="Search"
      className="form-control form-control-lg"
      placeholder="Search"
      type="search"
    />
  </form>
  <a
    className="nav-link"
    href="#"
  >
    Jane Doe
  </a>
</nav>
`;
